### ISSUE TITLE
proposal: Flag for disabling Earthly internal labels

### ISSUE THREAD

**Issue by 3manuek** [earthly/earthly#4069]
### Use case

Right now, Earthly includes a set of labels at `SAVE IMAGE` that are not related to the build itself:

```go
    labels := map[string]string{
        "dev.earthly.version":  version.Version,
        "dev.earthly.git-sha":  version.GitSha,
        "dev.earthly.built-by": version.BuiltBy,
    }
```

[Code](https://github.com/earthly/earthly/blob/2b6c37cc296ca1bede67e7a9a2a0e1c5fafb5423/earthfile2llb/interpreter.go#L1174-L1182)

A flag like `SAVE IMAGE --disable-earthly-labels` might work. Also, a feature flag might need to be added.

### Expected Behavior

It is expected that the user can control whether these labels are added or not to the produced image.

**Comment by alexcb** [comment]
2721c63ee6d923d6d801606913de47a90117c697 is a good example which shows what's needed to introduce a new feature flag.

**Comment by 3manuek** [comment]
I started the patch at #4084. I don't see how to make the version feature to work even tho I added it, probably I'm missing some compilation flag cc @alexcb .

**Comment by 3manuek** [comment]
This patch has been merged into [v0.8.10](https://github.com/earthly/earthly/releases/tag/v0.8.10). Closing, thanks @alexcb!
